src/platform/forms-system/src/js/components/SchemaForm.jsx

```diff
@@ -97,7 +101,20 @@ class SchemaForm extends React.Component {
   onBlur(id) {
     if (!this.state.formContext.touched[id]) {
       const formContext = _.set(['touched', id], true, this.state.formContext);
-      this.setState({ formContext });
+      // If we are running in a unit test environment, immediately update the
+      // state.
+      if (window.Mocha) {
+        this.setState({ formContext });
+      } else {
+        // Using a timeout of 100ms as that is long enough that a form can be
+        // removed from a parent if a "cancel" or "back" button is hit. This
+        // prevents any form validation errors from being shown for a fraction
+        // of a second before the form is removed. But we want to keep this
+        // delay short enough so that errors show up soon after leaving a field.
+        this.timeoutID = setTimeout(() => {
+          this.setState({ formContext });
+        }, 100);
```

Comment:
:wave: Just wanted to check in and let you know we're reviewing this. The FE Tools team is having a Slack discussion to see if there's a better way here. The `setTimeout` is...not our favorite. :sweat_smile: 

Have you considered alternatives to a timeout here? Or setting the timeout to 0? The concern is that timeouts are non-deterministic, so this isn't guaranteed to work _every_ time, and any bugs this may produce later are going to be a royal pain to debug, y'know?

Reply:
Yeah, totally fair. A timeout of 0 totally works; you just see the error message show up next to the offending field just as the form is being removed from the screen. So it fixes the bug, but adds a visual glitch or hiccup that might catch a user's eye. So 100ms was long enough that you don't see error messages but short enough that it doesn't negatively affect the UX under normal circumstances. Certainly far from ideal. We had two FE devs trying to fix the original bug and gave up twice. Then I thought of this setTimeout hack. I don't love it. But I also didn't want to dig even deeper into the form system and make larger changes.